app/services/aws/credentials_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Aws
+  class CredentialsService
+    class << self
+      def call(params = {})
+        new(params).call
+      end
+    end
```

Comment:
This is how ruby service objects convention goes. See for example this [article](https://dev.to/joker666/ruby-on-rails-pattern-service-objects-b19). If you see a way to make it nicer though, maybe elaborate a little more.

Reply:
I see, but it's not following the pattern because there's not an `ApplicationService` base class with the static call to be reused from all the services.

Reply:
There doesn't seem to be such a base class in Rails. Neither in porta codebase. I don't think it is a mandatory part of the pattern. It makes sense to have one and I'm also fine not to have if the duplication is just one method.

Also for me it is fine if `CMS::S3` is the single entrypoint to getting S3 configuration so this could be another helper class (not a service) that shouldn't be used by anything but `CMS::S3`. On the other hand STS credentials are potentially useful to other uses, not only S3 so it is also acceptable to be a service class.

Reply:
Regarding the service pattern, one thing I did in the past was to have a `Callable` Module that was added in whatever Service Object that didn't had the need to keep state (which was all of them, if memory serves right). It was something simple like:
```ruby
module Callable
  extend ActiveSupport::Concern

  included do
    private_class_method :new
  end

  class_methods do
    def call(*args)
      new(*args).call
    end
  end
end
```
Do you think it would make sense to add such a Module to Porta to be used on Service Objects?

Reply:
I think would be an improvement.

Reply:
@jlledom added in 20ea64e :v:

Reply:
Cool. The [article](https://dev.to/joker666/ruby-on-rails-pattern-service-objects-b19) also includes a `block` parameter.

```
class ApplicationService
  def self.call(*args, &block)
    new(*args, &block).call
  end
end
```
I think it's worth adding it, so we can handle all kinds of calls.